### ISSUE TITLE
ProfilingPlugin crashes build when using `source-map` devtool

### ISSUE THREAD

**Issue by RomanGotsiy** [webpack/webpack#6787]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
ProfilingPlugin crashes build when using `source-map` devtool

**If the current behavior is a bug, please provide the steps to reproduce.**

```js
const webpack = require('webpack');

module.exports = {
	devtool: 'source-maps',
	plugins: [
		new webpack.debug.ProfilingPlugin()
	]
}
```

`webpack --mode=production` fails with the following stack trace:

```
Unhandled rejection TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
    at compilation.hooks.afterOptimizeChunkAssets.tap (~/webpack-issue/node_modules/webpack/lib/SourceMapDevToolPlugin.js:108:26)
    at args (~/webpack-issue/node_modules/webpack/lib/debug/ProfilingPlugin.js:383:10)
    at SyncHook.eval [as call] (eval at create (~/webpack-issue/node_modules/tapable/lib/HookCodeFactory.js:17:12), <anonymous>:9:1)
    at SyncHook.lazyCompileHook [as _call] (~/webpack-issue/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.optimizeChunkAssets.callAsync.err (~/webpack-issue/node_modules/webpack/lib/Compilation.js:947:42)
    ...
```

Removing `devtool` fixes the build.

**What is the expected behavior?**
It should work

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

webpack 4.1.1
macOS High Siera 10.13.3
node v8.4.0

**Comment by chengluyu** [comment]
This will be fixed in #6788.

**Comment by ooflorent** [comment]
Duplicate of #6779
